src/controller/CHIPClusters.cpp

```diff
@@ -3691,7 +3701,10 @@ CHIP_ERROR NetworkCommissioningCluster::AddThreadNetwork(Callback::Cancelable *
 
     uint8_t argSeqNumber = 0;
     // operationalDataset: octetString
-    ReturnErrorOnFailure(writer.PutString(TLV::ContextTag(argSeqNumber++), operationalDataset));
+    // Check the size of the buffer, TLV::Write currently takes a uint32_t instead of size_t.
+    VerifyOrReturnError(operationalDataset.Length() <= UINT32_MAX, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
And here.

But really, shouldn't we just add a `PutBytes` overload on TLVWriter that takes this type, so we don't have to do this checking and the annoying separate Data() and Length() calls everywhere?